contracts/staking/SimpleStaking.sol

```diff
@@ -377,28 +429,10 @@ contract SimpleStaking is
 	function currentUBIInterest()
 		public
 		view
-		override
+		virtual
 		returns (uint256, uint256)
 	{
-		uint256 er = exchangeRate();
-		uint256 tokenWorth = currentTokenWorth();
-		if (tokenWorth <= totalProductivity) {
-			return (0, 0);
-		}
-		uint256 tokenGains = tokenWorth - totalProductivity;
-		(uint256 decimalDifference, bool caseType) = tokenDecimalPrecision();
-		//mul by `10^decimalDifference` to equalize precision otherwise since exchangerate is very big, dividing by it would result in 0.
-		uint256 iTokenGains;
-		uint256 mantissa = 18 + tokenDecimal() - iTokenDecimal(); // based on https://compound.finance/docs#protocol-math
-		if (caseType) {
-			iTokenGains =
-				((tokenGains / 10**decimalDifference) * 10**mantissa) /
-				er; // based on https://compound.finance/docs#protocol-math
-		} else {
-			iTokenGains =
-				((tokenGains * 10**decimalDifference) * 10**mantissa) /
-				er; // based on https://compound.finance/docs#protocol-math
-		}
+		(uint256 iTokenGains, uint256 tokenGains, ) = currentGains();
 		tokenGains = getTokenValueInUSD(tokenGains);
 		return (iTokenGains, tokenGains);
```

Comment:
should return triplet
itokengains,tokengains,usdgains

see also comment on interestcollected event

Reply:
changed it